### PR TITLE
Implement puppet-lint-class_parameter-check

### PR DESCRIPTION
**Pull request by bastelfreak**

### CODE REVIEW

**Comment by bastelfreak** [comment]
This is a test for a plugin. To get this working we require https://github.com/ryreitsma/puppet-lint-class_parameter-check/pull/25 to be fixed. @voxpupuli/collaborators please let me know if you like the idea that parameters are sorted alphabetically

**Comment by ekohl** [comment]
I generally don't like this because I'm worried about usefulness. For example, I like to keep `manage_db` together with `db_name`. Maybe it should be `db_manage` but renaming a parameter just so you can sort doesn't feel like a huge improvement.

File: manifests/plugin/aggregation.pp [hunk 1]
```diff
@@ -1,8 +1,8 @@
 # collectd::plugin::aggregation
 class collectd::plugin::aggregation (
-  Enum['present', 'absent'] $ensure = 'present',
-  Optional[Integer[1]] $interval    = undef,
   Hash $aggregators                 = {},
+  Enum['present', 'absent'] $ensure = 'present',
```
Comment (ekohl) [inline manifests/plugin/aggregation.pp:4]: I believe there's another lint check that prefers ensure to be the first param. Perhaps that's only for instantiation though.

Comment (bastelfreak) [reply manifests/plugin/aggregation.pp:4]: correct, that's only for resources (and as far as I know thats active in our setup)

**Comment by bastelfreak** [comment]
@ekohl I agree, personally I'm not 100% sure if I want this plugin or not, but I thought we gave it a try and let our collaborators decide.

**Comment by traylenator** [comment]
Parameters are often sorted by usefulness as well I'd say or relevance.
```
users_allow
groups_allow
```
you would not want to split.

**Comment by ekohl** [comment]
So what's the conclusion about this?

**Comment by ghoneycutt** [comment]
@bastelfreak I like this for modules with only a few parameters. The functionality of just checking that parameters without defaults go before parameters with defaults would be very useful without the rest of the params being in alphabetic ordering. 

Take for instance a class that has a template with a lot of parameters in it. It is more useful to separate the class params by function so you might have params in alpha order that all do with the module like file name, service name, etc though the params that are used in templateA are all together and the params for templateB are all together.

**Comment by ghoneycutt** [comment]
@bastelfreak Functionality aside, I'm not a fan of commits that re-order things or simply change their formatting, much like an aggressive rubocop auto fix. The reason is that it forces people to do some complicated rebases that could just turn them away from the project. If you are going to do a large amount of reformatting code, it is better to do when there are no big PR's in flight and to coincide with a major version bump. That will minimize the impact to developers.

**Comment by bastelfreak** [comment]
since not a majority voted for this plugin I'm going to close the PR.

**Pull request closed without merging**